### ISSUE TITLE
48.0 Infantry can jump through buildings contrary to TW rules

### ISSUE THREAD

**Issue by gsparks3** [MegaMek/megamek#2935]
### Description
Per Total Warfare p. 168, if an infantry unit is moving from a building hex to an adjacent building hex (including another hex of a multi-hex building), it must use Ground MP. Similarly, per TW p. 169, when changing levels inside a building, infantry must use Ground MP. 

They may jump _into_ a building hex (at a level below that of the roof) from outside the building, but their movement ends immediately as soon as they enter the first building hex, and they must have line of sight to said hex and elevation from their initial position. Jumping onto the roof is the same as jumping onto a clear hex at higher elevation and is not subject to such restrictions.

Currently, Megamek allows infantry with Jump MP to jump to any elevation of any building hex within their jump radius, regardless of how many floors or walls this involves jumping through. The desired hex is selected by clicking on it as normal, while the desired elevation is selected by using the "Go Up" and "Go Down" buttons. This should be adjusted to match the TW rules.

Reproduction steps:
1. Open a fresh Megamek lobby and select a map with buildings (16x17 Forward Base works well).
2. Add jump-capable infantry. I use Jump Platoon (Rifle) and Cavalier BA, but the type is not particularly important.
3. Experiment with jumping through said buildings in ways that violate the TW rules.

Examples: 

Jumping from outside the building to the ground floor of an interior hex
![image](https://user-images.githubusercontent.com/59574316/121624192-16405f80-ca3f-11eb-9d10-d3c911f9c938.png)
Jumping from the ground floor of one corner to the roof of the far corner
![image](https://user-images.githubusercontent.com/59574316/121624285-3e2fc300-ca3f-11eb-9e2d-df02f9a43bbe.png)
Jumping from the roof to the first floor of an adjacent hex
![image](https://user-images.githubusercontent.com/59574316/121624534-b5fded80-ca3f-11eb-867c-02ce44f1a9ac.png)

_Correct_ behavior would be: jumping from outside the building to any elevation of any exterior building hex that was within LoS of the initial position, jumping from inside the building to any point where the jump path does not pass through any other building hexes, or jumping onto / off of the roof normally. 

TLDR - you can jump into or out of a window but you shouldn't be able to jump through the floors, walls, or ceilings.

**Comment by gsparks3** [comment]
Additional note: jumping "into a window" (entering a building hex using Jump MP at a level lower than the roof from outside said building) should require an Anti-Mech Skill roll to avoid taking damage. This currently does not seem to be required by Megamek if the BA or jump infantry in question can reach the roof of the building, i.e. the first image above if the jump platoon were jumping into hex 0408 instead of 0410. It seems to be processed as if they jumped onto the roof and then teleported downwards the desired number of levels. However, if the roof of building is taller than the infantry can reach (e.g. the Jump Platoon jumping into a height 4 building from a hex on the same level), an AM skill roll will properly be required.
